crates/edr_provider/src/requests/hardhat/config.rs

```diff
@@ -0,0 +1,58 @@
+use edr_eth::{Address, SpecId, B256, U256};
+use rpc_hardhat::Metadata;
+
+use crate::{data::ProviderData, requests::eth::client_version, ProviderError};
+
+pub async fn handle_metadata_request(data: &ProviderData) -> Result<Metadata, ProviderError> {
+    Ok(Metadata {
+        client_version: client_version(),
+        chain_id: data.chain_id().await,
+        instance_id: *data.instance_id(),
+        latest_block_number: data.last_block_number().await,
+        latest_block_hash: *data.last_block().await?.hash(),
+        forked_network: data.fork_metadata().cloned(),
+    })
+}
+
+pub fn handle_set_coinbase_request(
+    data: &mut ProviderData,
+    coinbase: Address,
+) -> Result<bool, ProviderError> {
+    data.set_coinbase(coinbase);
+
+    Ok(true)
+}
+
+pub fn handle_set_next_block_base_fee_per_gas_request(
+    data: &mut ProviderData,
+    base_fee_per_gas: U256,
+) -> Result<bool, ProviderError> {
+    let spec_id = data.spec_id();
+    if spec_id < SpecId::LONDON {
+        return Err(ProviderError::UnmetHardfork {
```

Comment:
Wondering if this is going to match error returned by Hardhat. Is that a concern at all @alcuadrado @fvictorio ?